core/src/test/java/org/axonframework/queryhandling/SubscriptionQueryTest.java

```diff
@@ -50,183 +49,302 @@ public void setUp() {
     }
 
     @Test
-    public void testRegularQueryWithSubscriptionQueryHandler() throws ExecutionException, InterruptedException {
+    public void testEmittingAnUpdate() {
         // given
-        QueryMessage<String, List<String>> queryMessage =
-                new GenericQueryMessage<>("axonFrameworkCR",
-                                          "subscribableChatMessages",
-                                          ResponseTypes.multipleInstancesOf(String.class));
+        SubscriptionQueryMessage<String, List<String>, String> queryMessage1 = new GenericSubscriptionQueryMessage<>(
+                "axonFrameworkCR",
+                "chatMessages",
+                ResponseTypes.multipleInstancesOf(String.class),
+                ResponseTypes.instanceOf(String.class));
+
+        SubscriptionQueryMessage<Integer, Integer, Integer> queryMessage2 = new GenericSubscriptionQueryMessage<>(
+                5,
+                "numberOfMessages",
+                ResponseTypes.instanceOf(Integer.class),
+                ResponseTypes.instanceOf(Integer.class));
 
         // when
-        List<String> messages = queryBus.query(queryMessage).thenApply(Message::getPayload).get();
+        SubscriptionQueryResult<QueryResponseMessage<List<String>>, SubscriptionQueryUpdateMessage<String>> result1 = queryBus
```

Comment:
Although I understand the contained generics of the `SubscriptionQueryResult` are a `QueryResponseMessage` and `SubscriptionQueryUpdateMessage`, from a users perspective I'd say it would be cleaner if we could obscure that from the users.

Reply:
Last night I was kinda in doubt whether this holds in all cases I could think of.

I was generalizing our bus' and gateways when I said previous comment.
Our gateways abstract the knowledge of the `...Message` implementations, thus allowing users to provide and retrieve the payload. Our bus' however still require the `...Message`  as far as I know. 

As you're using the `QueryBus` in this test, I shouldn't have jumped the gun as quick as I did. 
In short, disregard my comment @m1l4n54v1c

Reply:
Yes, we have `QueryGateway` for that purposes.